src/net/sourceforge/zmanim/AstronomicalCalendar.java

```diff
@@ -143,15 +145,15 @@ public Date getSunrise() {
 	 * adjustment}. Non-sunrise and sunset calculations such as dawn and dusk, depend on the amount of visible light,
 	 * something that is not affected by elevation. This method returns sunrise calculated at sea level. This forms the
 	 * base for dawn calculations that are calculated as a dip below the horizon before sunrise.
-	 * 
-	 * @return the <code>Date</code> representing the exact sea-level sunrise time. If the calculation can't be computed
+	 *
+	 * @return the date representing the exact sea-level sunrise time. If the calculation can't be computed
 	 *         such as in the Arctic Circle where there is at least one day a year where the sun does not rise, and one
 	 *         where it does not set, a null will be returned. See detailed explanation on top of the page.
 	 * @see AstronomicalCalendar#getSunrise
 	 * @see AstronomicalCalendar#getUTCSeaLevelSunrise
 	 * @see #getSeaLevelSunset()
 	 */
-	public Date getSeaLevelSunrise() {
+	public Long getSeaLevelSunrise() {
 		double sunrise = getUTCSeaLevelSunrise(GEOMETRIC_ZENITH);
 		if (Double.isNaN(sunrise)) {
 			return null;
```

Comment:
Actually, I was wondering if this method should be changed to return NEVER *instead of* null.

No doubt, however, that being an API breaking change, is actually a bad idea. So...never mind ;-)